src/components/extexplorer/explorer.ts

```diff
@@ -0,0 +1,73 @@
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Kubectl } from '../../kubectl';
+
+interface Recommendations {
+    [crdApi: string]: string[];
+}
+
+async function recommendExtensionsToUser(tags: string[]): Promise<void> {
+    if (tags.length > 0) { // do not prompt if we cannot recommend anything
+        const answer = await vscode.window.showInformationMessage('Do you want to view recommended extensions for your Kubernetes cluster?', 'Ignore', 'Show Recommendation');
+        if (answer === 'Show Recommendation') {
+            const searchValue = tags.map((it) => `@tag:${it}`).join(' ');
```

Comment:
Yes, that search by tag.
In https://github.com/Azure/vscode-kubernetes-tools/pull/780#issuecomment-778855475 ,
You propose to provide recommendation with search, instead of recommending specific extensions:

>We also need to consider how to evolve the recommendations database. I don't want to become responsible for evaluating, say, multiple Tekton extensions to decide which is "recommended." Is our bar going to be "anyone can put in a PR recommendingy any extension and as long as it's not obviously evil then we'll give it our blessing"? This feels like it risks our reputation (consider a malicious actor who creates a plausible extension, gets us to recommend it, and then modifies it to steal data). A plain "search for extensions for CRD X" based on a keyword is more neutral than "we recommend" and would require no curation - but might provide correspondingly less value.

Maybe I miss understand your point.

Reply:
No, I had forgotten this discussion I'm afraid!  Thanks for reminding me!  I agree with past Ivan - tag search avoids maintenance and evaluation issues.  But in that case we _must_ say something like "*search for* extensions for your CRDs" not "recommended extensions" - recommended implies endorsement and curation, search implies only "we'll save you typing in the keyword yourself."

So, having been reminded of those thoughts - I now agree with what you are doing, but would like to see the UI at least changed from "show recommended extensions" to "search for extensions".

With that change we will be good to go I think!

Reply:
@evidolob you missed this point. We also need to change the statement `'Do you want to view recommended extensions for your Kubernetes cluster?'`. By citing Ivan "recommended implies endorsement and curation".
Maybe `Do you want to search for extensions based on CRDs that exist on your kubernetes cluster?"

Reply:
OK, I change that message. Thx @lstocchi !